src/main/java/org/jvnet/hudson/test/HudsonTestCase.java

```diff
@@ -1007,7 +1005,7 @@ public Object call() throws Exception {
      */
     private List<String> listProperties(String properties) {
         List<String> props = new ArrayList<String>(Arrays.asList(properties.split(",")));
-        for (String p : props.toArray(new String[props.size()])) {
+        for (String p : props.toArray(new String[0])) {
```

Comment:
This looks quite wrong to me, care to explain?

Reply:
The commit comment contains the explanation. It also refers to https://shipilev.net/blog/2016/arrays-wisdom-ancients/ with a very detailed analysis.

Reply:
That applies to copying arrays. In this line, you are only iterating over them, so you don't need to copy. It should just be:

```suggestion
        for (String p : props) {
```

Reply:
@jvz your suggested change is not applicable as it would result in a `ConcurrentModificationException` when removing elements from the list while iterating over it.

Reply:
What if you made props a `CopyOnWriteArrayList` instead?

Reply:
That's definitely an option. I updated the original commit and simplied it even further - see 288464dffc8f1d3f0708287d9ba6bc3f0f4a7d5a.